model/src/main/java/com/basistech/rosette/dm/TranslatedData.java

```diff
@@ -83,18 +111,33 @@ protected MoreObjects.ToStringHelper toStringHelper() {
      * Builder class for TranslatedData.
      */
     public static class Builder extends BaseAttribute.Builder<TranslatedData, TranslatedData.Builder> {
-        private TextDomain domain;
-        private String translation;
+        private final TextDomain sourceDomain;
+        private final TextDomain targetDomain;
+        private final String translation;
         private Double confidence;
 
         /**
          * Constructs a builder from the required properties
          *
-         * @param domain specifies the language and script of the translation
+         * @param targetDomain specifies the language and script of the translation
+         * @param translation the translation for the text
+         */
+        public Builder(TextDomain targetDomain, String translation) {
```

Comment:
Like the main class constructor, this should be deprecated in favor of the one that also takes in a sourceDomain